src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStep.java

```diff
@@ -0,0 +1,187 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.utility.steps.conf;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.pipeline.utility.steps.shaded.org.yaml.snakeyaml.DumperOptions;
+import org.jenkinsci.plugins.pipeline.utility.steps.shaded.org.yaml.snakeyaml.Yaml;
+import org.jenkinsci.plugins.workflow.steps.*;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.io.*;
+import java.net.URL;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.*;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+/**
+ * Writes a yaml file from the workspace.
+ *
+ * @author Javier DELGADO &lt;witokondoria@gmail.com&gt;.
+ */
+public class WriteYamlStep extends AbstractStepImpl {
+
+    private String file;
+    private Object data;
+
+    @DataBoundConstructor
+    public WriteYamlStep(@Nonnull String file, @Nonnull Object data) {
+        if (file == null) {
+            throw new IllegalArgumentException("file parameter must be provided to writeYaml");
+        }
+        this.file = file;
+        if (data == null) {
+            throw new IllegalArgumentException("data parameter must be provided to writeYaml");
+        } else if (!isValidObjectType(data)) {
+            throw new IllegalArgumentException("data parameter has invalid content (no-basic classes)");
+        }
+        this.data = data;
+    }
+
+    /**
+     * Name of the yaml file to write.
+     *
+     * @return file name
+     */
+    public String getFile() {
+        return file;
+    }
+
+    /**
+     * Name of the yaml file to write.
+     *
+     * @param file file name
+     */
+    public void setFile(String file) {
+        this.file = file;
+    }
+
+    /**
+     * An Object containing data to be saved.
+     *
+     * @return data to save as yaml
+     */
+    public Object getData() {
+        return data;
+    }
+
+    /**
+     * An Object representing the elements to write.
+     *
+     * @param data to parse
+     */
+    public void setData(Object data) {
+        this.data = data;
+    }
+
+    private boolean isValidObjectType(Object obj) {
+        if ((obj instanceof Boolean) || (obj instanceof Character) ||
+            (obj instanceof Number) || (obj instanceof String) ||
+            (obj instanceof URL) || (obj instanceof Calendar) ||
+            (obj instanceof Date) || (obj instanceof UUID) ||
+            (obj == null)) {
+            return true;
+        } else if (obj instanceof Map)  {
+            Boolean ret = true;
+            Set entry = ((Map) obj).entrySet();
+            Iterator iterator = entry.iterator();
+            while (iterator.hasNext() && ret) {
```

Comment:
This looks a bit convoluted, something like this would be more readable and simpler I think

```java
for (Map.Entry entry : map.entrySet()) {
    if (!isValidObject(entry.getKey()) || !isValidObject(entry.getValue())) {
        return false;
    }
}
```

It also seems like you are overwriting the `ret` var every time in the loop, so effectively you are only checking the last entry in the map which my suggestion also solves.